File: data/src/main/java/com/crhonvas/data/remote/api/users/IUserApi.java
package com.crhonvas.data.remote.api.users;

import com.crhonvas.data.remote.model.user.UserApiResponse;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface IUserApi {
    @GET("users")
    Single<List<UserApiResponse>> getUsers();

    @GET("users/{id}")
    Single<UserApiResponse> getUser(@Path("id") Integer id);
}

File: data/src/main/java/com/crhonvas/data/remote/api/users/IUserApi.kt
package com.crhonvas.data.remote.api.users

import com.crhonvas.data.remote.model.user.UserApiResponse
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface IUserApi {
    @get:GET("users")
    val users: Single<List<UserApiResponse?>?>?

    @GET("users/{id}")
    fun getUser(@Path("id") id: Int?): Single<UserApiResponse>
}